lib/floe/workflow/state.rb

```diff
@@ -48,17 +48,27 @@ def run_nonblock!(context)
         return Errno::EAGAIN unless ready?(context)
 
         finish(context)
+      rescue Floe::ExecutionError => e
+        mark_error(context, e)
```

Comment:
This was detected because tests running against a state would work, but running against a workflow would not.

Before:
- Errors needed to be caught by each `State`
- Errors resulted in a state that was not complete.
- Errors did not always get their way into output.

After:
- runtime errors always set the context accordingly.